Runtime/AvatarCreator/UI/SelfieElement.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using ReadyPlayerMe.Core;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+    public class SelfieElement : MonoBehaviour
+    {
+        private const string TAG = nameof(SelfieElement);
+
+        [SerializeField] private ImageConfirmationElement imageConfirmationElement;
+        [SerializeField] private BodyType bodyType;
+        [SerializeField] private OutfitGender gender;
+
+        [Space(5)]
+        [Header("Events")]
+        public UnityEvent<GameObject,AvatarProperties> onAvatarCreated;
+
+        private void OnEnable()
+        {
+            imageConfirmationElement.onImageConfirmed.AddListener(OnPhotoCaptured);
+        }
+
+        private void OnDisable()
+        {
+            imageConfirmationElement.onImageConfirmed.RemoveListener(OnPhotoCaptured);
+        }
+
+        private async void OnPhotoCaptured(Texture2D texture)
+        {
+            var bytes = texture.EncodeToPNG();
+            var byteAsString = Convert.ToBase64String(bytes);
+
+            var avatarProperties = new AvatarProperties();
+            avatarProperties.Partner = CoreSettingsHandler.CoreSettings.Subdomain;
+            avatarProperties.Base64Image = byteAsString;
+            avatarProperties.BodyType = bodyType;
+            avatarProperties.Gender = gender;
+            avatarProperties.Assets = new Dictionary<Category, object>();
+
+            if (!AuthManager.IsSignedIn && !AuthManager.IsSignedInAnonymously)
+            {
+                SDKLogger.Log(TAG, "Not signed in");
+                return;
+            }
+            
+            var avatarManager = new AvatarManager(bodyType);
```

Comment:
Minor thing but I just realized that bodyType is being set in multiple places. I asume that the setting in the constructor will override the AvatarProps one. But probably that may be confusing to devs.
Perhaps we should just remove bodyType from the constructor, what do you think?

Reply:
Yes we could do that, but than we have to provide almost all of the functions of AvatarManager with bodyType.

Reply:
What do you mean exactly by having to provide all functions to AvatarManager?